openfpga/src/utils/report_reference.cpp

```diff
@@ -0,0 +1,163 @@
+/***************************************************************************************
+ * Output internal structure of module graph to XML format
+ ***************************************************************************************/
+/* Headers from system goes first */
+#include <algorithm>
+#include <chrono>
+#include <ctime>
+#include <iomanip>
+#include <string>
+
+/* Headers from vtrutil library */
+#include "vtr_assert.h"
+#include "vtr_log.h"
+#include "vtr_time.h"
+
+/* Headers from openfpgautil library */
+#include "command_exit_codes.h"
+#include "openfpga_digest.h"
+#include "openfpga_naming.h"
+#include "report_reference.h"
+
+/* begin namespace openfpga */
+namespace openfpga {
+
+/********************************************************************
+ * Top-level function
+ *******************************************************************/
+int report_reference(const char* fname, const std::string& module_name,
+                     const ModuleManager& module_manager,
+                     const bool& include_time_stamp, const bool& verbose) {
+  vtr::ScopedStartFinishTimer timer("Report reference");
+
+  ModuleId parent_module = module_manager.find_module(module_name);
+  if (false == module_manager.valid_module_id(parent_module)) {
+    VTR_LOG_ERROR("Module %s doesn't exist\n", module_name.c_str());
+
+    if (verbose) write_module_to_file(fname, module_manager);
```

Comment:
Is this for debug purpose? It may not be an expected output. If we do need this, a message should be added.

Reply:
Yes. it's for debug. And it's just to resolve  unused warning of the argument verbose.  If it is not used, an unused warning will occur in PR checking process.  For the time being, I am not sure what is necessary to do with verbose argument.  Could you give me a more detailed suggestion on what to do for verbose argument? Thanks.

Reply:
I perfer to remove LINE 37. Since you have the verbose option used at LINE 47. You should not see the warning on unused variables ?

Reply:
OK, I have removed LINE 37.